contracts/LivepeerProtocol.sol

```diff
@@ -523,11 +540,87 @@ contract LivepeerProtocol {
     }
 
     /*
-     * Pseudorandomly elect an active transcoder. Currently a placeholder
-     * TODO: take into account pricing, etc.
+     * Submit a transcoding job
+     * @param _streamId Unique stream identifier
+     * @param _transcodingOptions Output bitrates, formats, encodings
+     * @param _maxPricePerSegment Max price (in LPT base units) to pay for transcoding a segment of a stream
+     */
+    function job(uint256 _streamId, bytes32 _transcodingOptions, uint256 _maxPricePerSegment) returns (bool) {
+        address electedTranscoder = electCurrentActiveTranscoder(_maxPricePerSegment);
+
+        // Check if there is an elected current active transcoder
+        if (electedTranscoder == address(0)) throw;
+
+        jobs.push(Job(jobs.length - 1, _streamId, _transcodingOptions, _maxPricePerSegment, msg.sender, electedTranscoder, 0));
+
+        return true;
+    }
+
+    /*
+     * Computes the status of a job
+     * @param _jobId Job identifier
+     */
+    function jobStatus(uint256 _jobId) constant returns (JobStatus) {
+        if (jobs[_jobId].endBlock > 0 && jobs[_jobId].endBlock <= block.number) {
+            // A job is inactive if its end block is set and the current block is greater than or equal to the job's end block
+            return JobStatus.Inactive;
+        } else {
+            // A job is active if the current block is less than the job's termination block
+            return JobStatus.Active;
+        }
+    }
+
+    /*
+     * Return a job
+     * @param _jobId Job identifier
+     */
+    function getJob(uint256 _jobId) constant returns (uint256, uint256, bytes32, uint256, address, address, uint256) {
+        Job job = jobs[_jobId];
+
+        return (job.jobId, job.streamId, job.transcodingOptions, job.maxPricePerSegment, job.broadcasterAddress, job.transcoderAddress, job.endBlock);
+    }
+
+    /*
+     * End a job. Can be called by either a broadcaster or transcoder of a job
+     */
+    function endJob(uint256 _jobId) returns (bool) {
```

Comment:
In one of the issues `endJob()` is described as the function called by transcoders to submit transcode claims for a range of segments. This name does not seem too semantically accurate and I think `endJob()` would make more sense as a name for a function used by a broadcaster or transcoder to set a job as inactive. Perhaps the function for submitting transcode claims for a range of segments can instead be called `claimWorkForJob()`.